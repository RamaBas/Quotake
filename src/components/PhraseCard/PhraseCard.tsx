import React from 'react';
import { Card, CardContent, IconButton, Typography, Box } from '@mui/material';
import { Close, FormatQuote } from '@mui/icons-material';

interface PhraseCardProps {
  phrase: {
    id: string;
    text: string;
  };
  onDelete: (id: string) => void;
  backgroundColor: string;
}

const PhraseCard: React.FC<PhraseCardProps> = ({ phrase, onDelete, backgroundColor }) => {
  return (
    <Card sx={{ 
      backgroundColor,
      position: 'relative',
      minHeight: 100,
      maxHeight: "auto",
      '&:hover': {
        boxShadow: 3,
      },
    }}>
      <Box
        sx={{
          position: 'absolute',
          top: 2,
          left: 2,
          zIndex: 0,
          opacity: 0.2,
        }}
      >
        <FormatQuote 
          sx={{ 
            fontSize: '3rem',
            color: backgroundColor === '#EDF7FA' ? 'text.primary' : '#EDF7FA',
          }} 
        />
      </Box>
      <CardContent sx={{ 
        position: 'relative', 
        zIndex: 1,
        flexGrow: 1,
        minHeight: '100%',
        display: 'flex',
        flexDirection: 'row',
      }}>
        <Typography 
          variant="body1" 
          sx={{ 
            padding: 2,
            wordBreak: 'break-word',
            flexGrow: 1,
          }}
        >
          {phrase.text}
        </Typography>
      </CardContent>

      <IconButton
        aria-label="Eliminar frase"
        onClick={() => onDelete(phrase.id)}
        sx={{
          position: 'absolute',
          top: 2,
          right: 2,
          zIndex: 2, // Encima de todo
          '&:hover': {
            color: 'error.main',
            backgroundColor: 'rgba(220, 0, 78, 0.1)',
          },
        }}
      >
        <Close fontSize="small" />
      </IconButton>
    </Card>
  );
};

export default PhraseCard;