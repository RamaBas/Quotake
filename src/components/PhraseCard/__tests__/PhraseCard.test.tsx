import { render } from '@testing-library/react';
import PhraseCard from '../PhraseCard';

// Mock para los iconos de MUI
jest.mock('@mui/icons-material', () => ({
  Close: () => <div>CloseIcon</div>,
  Undo: () => <div>UndoIcon</div>,
  FormatQuote: () => <div>FormatQuoteIcon</div>,
}));

describe('PhraseCard Component Snapshots', () => {
  const mockPhrase = {
    id: '1',
    text: 'Esta es una frase de prueba para el componente'
  };

  it('matches default variant snapshot', () => {
    const { container } = render(
      <PhraseCard 
        phrase={mockPhrase}
        onDelete={jest.fn()}
        variant="default"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches deleted variant snapshot', () => {
    const { container } = render(
      <PhraseCard 
        phrase={mockPhrase}
        onRestore={jest.fn()}
        variant="deleted"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot without actions', () => {
    const { container } = render(
      <PhraseCard 
        phrase={mockPhrase}
        variant="default"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});