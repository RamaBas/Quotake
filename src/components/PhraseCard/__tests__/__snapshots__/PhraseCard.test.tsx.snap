// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhraseCard Component Snapshots matches default variant snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-152yur6-MuiPaper-root-MuiCard-root"
  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
>
  <div
    class="MuiBox-root css-oy3rny"
  >
    <div>
      FormatQuoteIcon
    </div>
  </div>
  <div
    class="MuiCardContent-root css-rnega1-MuiCardContent-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1salclq-MuiTypography-root"
    >
      Esta es una frase de prueba para el componente
    </p>
  </div>
  <button
    aria-label="Eliminar frase"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-2wse5v-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <div>
      CloseIcon
    </div>
  </button>
</div>
`;

exports[`PhraseCard Component Snapshots matches deleted variant snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-16b9zzk-MuiPaper-root-MuiCard-root"
  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
>
  <div
    class="MuiBox-root css-oy3rny"
  >
    <div>
      FormatQuoteIcon
    </div>
  </div>
  <div
    class="MuiCardContent-root css-rnega1-MuiCardContent-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1salclq-MuiTypography-root"
    >
      Esta es una frase de prueba para el componente
    </p>
  </div>
  <button
    aria-label="Restaurar frase"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-vrkvbn-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <div>
      UndoIcon
    </div>
  </button>
</div>
`;

exports[`PhraseCard Component Snapshots matches snapshot without actions 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-152yur6-MuiPaper-root-MuiCard-root"
  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
>
  <div
    class="MuiBox-root css-oy3rny"
  >
    <div>
      FormatQuoteIcon
    </div>
  </div>
  <div
    class="MuiCardContent-root css-rnega1-MuiCardContent-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1salclq-MuiTypography-root"
    >
      Esta es una frase de prueba para el componente
    </p>
  </div>
</div>
`;
